Compile fails, undefined reference to `QgsServerFilter::QgsServerFilter(QgsServerInterface*)'

### What is the bug or the crash?

Building on RHEL7 with GCC 10.2.1 fails here (also tried devtoolset-8 and 9, no difference)

```
[ 41%] Linking CXX shared module ../../../../output/lib/qgis/server/liblandingpage.so
/opt/rh/devtoolset-10/root/usr/libexec/gcc/x86_64-redhat-linux/10/ld: CMakeFiles/landingpage.dir/qgslandingpage.cpp.o: in function `QgsProjectLoaderFilter::QgsProjectLoaderFilter(QgsServerInterface*)':
/home/wms/src/qgis-3.16.9-210812/src/server/services/landingpage/qgslandingpage.cpp:71: undefined reference to `QgsServerFilter::QgsServerFilter(QgsServerInterface*)'
/opt/rh/devtoolset-10/root/usr/libexec/gcc/x86_64-redhat-linux/10/ld: CMakeFiles/landingpage.dir/qgslandingpage.cpp.o:(.data.rel.ro._ZTI22QgsProjectLoaderFilter[_ZTI22QgsProjectLoaderFilter]+0x10): undefined reference to `typeinfo for QgsServerFilter'
/opt/rh/devtoolset-10/root/usr/libexec/gcc/x86_64-redhat-linux/10/ld: CMakeFiles/landingpage.dir/qgslandingpage.cpp.o:(.data.rel.ro._ZTV22QgsProjectLoaderFilter[_ZTV22QgsProjectLoaderFilter]+0x30): undefined reference to `QgsServerFilter::sendResponse()'
collect2: error: ld returned 1 exit status
```


### Steps to reproduce the issue

```
cmake .. \
  -DCMAKE_BUILD_TYPE=Release \
  -DWITH_SERVER=TRUE \
  -DWITH_SERVER_PLUGINS=FALSE \
  -DWITH_GUI=FALSE \
  -DWITH_DESKTOP=FALSE \
  -DWITH_POSTGRESQL=FALSE \
  -DWITH_BINDINGS=FALSE \
  -DWITH_QTWEBKIT=FALSE \
  -DWITH_QT5SERIALPORT=FALSE \
  -DENABLE_TESTS=FALSE
make
```

### Versions

Latest 3.16.9+git

### Additional context

_No response_

Here's the full compile command as shown by make VERBOSE=1

```
cd /home/wms/src/qgis-3.16.11/build/src/server/services/landingpage
/opt/rh/devtoolset-10/root/usr/bin/c++ -fPIC -fstack-protector-strong -g -O2 -DNDEBUG -Wall -Wextra -Wno-long-long -Wformat-security -Wno-strict-aliasing -Wnon-virtual-dtor -Wno-redundant-move -Wno-deprecated-copy -fvisibility=hidden -fvisibility-inlines-hidden -O2 -DNDEBUG -Wl,--rpath,/home/wms/qgis-3.16.11/lib,--rpath,/home/wms/geos-3.9.1/lib,--rpath,/home/wms/gdal-3.3.1/lib,--rpath,/home/wms/proj-8.1.0/lib,--rpath,/home/wms/sqlite-3360000/lib,--rpath,/home/wms/libspatialite-5.0.1/lib,--rpath,/home/wms/protobuf-3.17.3/lib,--rpath,/home/wms/libpng-1.6/lib -Wl,--no-undefined -shared  -o ../../../../output/lib/qgis/server/liblandingpage.so CMakeFiles/landingpage.dir/landingpage_autogen/mocs_compilation.cpp.o CMakeFiles/landingpage.dir/qgslandingpage.cpp.o CMakeFiles/landingpage.dir/qgslandingpageutils.cpp.o CMakeFiles/landingpage.dir/qgslandingpagehandlers.cpp.o   -L/home/wms/src/qgis-3.16.11/build/src/core  -L/home/wms/src/qgis-3.16.11/build/src/gui  -Wl,-rpath,/home/wms/src/qgis-3.16.11/build/src/core:/home/wms/src/qgis-3.16.11/build/src/gui:/home/wms/src/qgis-3.16.11/build/output/lib:/home/wms/proj-8.1.0/lib:/home/wms/geos-3.9.1/lib:/home/wms/gdal-3.3.1/lib:/home/wms/sqlite-3360000/lib:/home/wms/libspatialite-5.0.1/lib:/home/wms/protobuf-3.17.3/lib: ../../../../output/lib/libqgis_server.so.3.16.11 /usr/lib64/libfcgi.so ../../../../output/lib/libqgis_analysis.so.3.16.11 ../../../../output/lib/libqgis_core.so.3.16.11 /usr/lib64/libQt5Xml.so.5.9.7 /usr/lib64/libQt5Svg.so.5.9.7 /usr/lib64/libQt5PrintSupport.so.5.9.7 /usr/lib64/libQt5Widgets.so.5.9.7 /usr/lib64/libQt5Gui.so.5.9.7 /usr/lib64/libQt5Network.so.5.9.7 /usr/lib64/libQt5Sql.so.5.9.7 /usr/lib64/libQt5Concurrent.so.5.9.7 /usr/lib64/libqca-qt5.so /usr/lib64/libqt5keychain.so /home/wms/proj-8.1.0/lib/libproj.so /home/wms/geos-3.9.1/lib/libgeos_c.so /home/wms/gdal-3.3.1/lib/libgdal.so /usr/lib64/libspatialindex.so /lib64/libexpat.so /home/wms/sqlite-3360000/lib/libsqlite3.so /home/wms/libspatialite-5.0.1/lib/libspatialite.so /usr/lib64/libzip.so /home/wms/protobuf-3.17.3/lib/libprotobuf-lite.so /usr/lib64/libz.so /usr/lib64/libQt5Positioning.so.5.9.7 /usr/lib64/libQt5Core.so.5.9.7 /usr/lib64/libexiv2.so
```

```
/opt/rh/devtoolset-10/root/usr/libexec/gcc/x86_64-redhat-linux/10/ld: CMakeFiles/landingpage.dir/qgslandingpage.cpp.o: in function `QgsProjectLoaderFilter::QgsProjectLoaderFilter(QgsServerInterface*)':
/home/wms/src/qgis-3.16.11/src/server/services/landingpage/qgslandingpage.cpp:71: undefined reference to `QgsServerFilter::QgsServerFilter(QgsServerInterface*)'
/opt/rh/devtoolset-10/root/usr/libexec/gcc/x86_64-redhat-linux/10/ld: CMakeFiles/landingpage.dir/qgslandingpage.cpp.o:(.data.rel.ro._ZTI22QgsProjectLoaderFilter[_ZTI22QgsProjectLoaderFilter]+0x10): undefined reference to `typeinfo for QgsServerFilter'
/opt/rh/devtoolset-10/root/usr/libexec/gcc/x86_64-redhat-linux/10/ld: CMakeFiles/landingpage.dir/qgslandingpage.cpp.o:(.data.rel.ro._ZTV22QgsProjectLoaderFilter[_ZTV22QgsProjectLoaderFilter]+0x30): undefined reference to `QgsServerFilter::sendResponse()'
```

Regardless of using RHEL or not, perhaps someone familiar with QGIS internals can tell me what is missing.

QgsServerFilter::sendResponse() is found in src/server/qgsserverfilter.cpp, so is it not included in libqgis_server.so for some reason or something?

--- a/src/server/services/wmts/qgswmtsutils.h
+++ b/src/server/services/wmts/qgswmtsutils.h
@@ -23,6 +23,8 @@
 #include "qgswmtsparameters.h"
 #include "qgswmtsserviceexception.h"
 #include "qgsserversettings.h"
+#include "qgsunittypes.h"
+#include "qgsrectangle.h"
 
 #include <QDomDocument>
 

